examples/window/expanding_window.rs

```diff
@@ -0,0 +1,106 @@
+use bevy::prelude::*;
+
+const MAX_WIDTH: f32 = 400.;
+const MAX_HEIGHT: f32 = 400.;
+
+fn main() {
+    App::new()
+        .insert_resource(WindowDescriptor {
+            width: MAX_WIDTH,
+            height: MAX_HEIGHT,
+            scale_factor_override: Some(1.),
+            ..Default::default()
+        })
+        .add_plugins(DefaultPlugins)
+        .insert_resource(Phase::ContractingY)
+        .add_system(change_window_size)
+        .add_startup_system(setup)
+        .run();
+}
+
+enum Phase {
+    ContractingY,
+    ContractingX,
+    ExpandingY,
+    ExpandingX,
+}
+
+use Phase::*;
+
+fn change_window_size(
+    mut windows: ResMut<Windows>,
+    mut phase: ResMut<Phase>,
+    mut first_complete: Local<bool>,
+) {
+    // Put off rendering for one frame, as currently for a frame where
+    // resizing happens, nothing is presented.
+    if !*first_complete {
+        *first_complete = true;
+        return;
+    }
+    let primary = windows.get_primary_mut().unwrap();
+    let height = primary.height();
+    let width = primary.width();
+    match *phase {
+        Phase::ContractingY => {
+            if height <= 0.5 {
+                *phase = ContractingX;
+            }
+            primary.set_resolution(width, (height - 4.).max(0.0))
```

Comment:
```suggestion
            primary.set_resolution(width, (height - 4.).max(1.0))
```
maybe this here and below works? May also need to be a couple of pixels bigger to account for window decorations.

Reply:
New version should avoid this